inference-engine/src/vpu/common/src/ngraph/operations/dynamic_shape_resolver.cpp

```diff
@@ -30,18 +34,162 @@ void DynamicShapeResolver::validate_and_infer_types() {
+    }
 }
 
 bool DynamicShapeResolver::visit_attributes(ngraph::AttributeVisitor&) {
     return true;
 }
 
+namespace {
+
+template<element::Type_t ET>
+bool getShapeFromHostTensorData(const HostTensorPtr& data, Shape& result) {
+    using T = typename element_type_traits<ET>::value_type;
+    T *dataPtr = data->get_data_ptr<ET>();
+    if (!dataPtr) {
+        return false;
+    }
+    if (data->get_shape().size() != 1) {
+        return false;
+    }
+    size_t outputRank = data->get_shape()[0];
+
+    for (int i = 0; i < outputRank; i++) {
+        result.push_back(dataPtr[i]);
+    }
+
+    return true;
+}
+
+bool getShapeFromHostTensorData(const HostTensorPtr& data, Shape& shape) {
+    switch (data->get_element_type()) {
+        case element::Type_t::i8:
+            return getShapeFromHostTensorData<element::Type_t::i8>(data, shape);
+        case element::Type_t::i16:
+            return getShapeFromHostTensorData<element::Type_t::i16>(data, shape);
+        case element::Type_t::i32:
+            return getShapeFromHostTensorData<element::Type_t::i32>(data, shape);
+        case element::Type_t::i64:
+            return getShapeFromHostTensorData<element::Type_t::i64>(data, shape);
+        case element::Type_t::u8:
+            return getShapeFromHostTensorData<element::Type_t::u8>(data, shape);
+        case element::Type_t::u16:
+            return getShapeFromHostTensorData<element::Type_t::u16>(data, shape);
+        case element::Type_t::u32:
+            return getShapeFromHostTensorData<element::Type_t::u32>(data, shape);
+        case element::Type_t::u64:
+            return getShapeFromHostTensorData<element::Type_t::u64>(data, shape);
+        default:
+            return false;
+    }
+    return true;
+}
+
+template<element::Type_t InType>
+bool evaluate(const HostTensorPtr& inputTensor,
+              const HostTensorPtr& inputShapeTensor,
+              const HostTensorPtr& outputTensor) {
+    Shape inputShape = inputTensor->get_shape();
+    Shape outputShape;
+    if (!getShapeFromHostTensorData(inputShapeTensor, outputShape)) {
+        return false;
+    }
+
+    if (!ngraph::PartialShape(outputShape).refines(outputTensor->get_partial_shape())) {
+        return false;
+    }
+
+    outputTensor->set_shape(outputShape);
+
+    using T = typename element_type_traits<InType>::value_type;
+    T *inputPtr = inputTensor->get_data_ptr<InType>();
+    T *outputPtr = outputTensor->get_data_ptr<InType>();
+
+    const auto inTotalDimSize = shape_size(inputShape);
+    const auto strides = row_major_strides(inputShape);
+
+    for (size_t inputTensorIdx = 0, outputTensorIdx = 0; inputTensorIdx < inTotalDimSize; ++inputTensorIdx) {
+        auto offset = inputTensorIdx;
+        bool needCopy = true;
+        for (size_t dim = 0; dim < strides.size(); ++dim) {
+            const auto coordAlongDim = offset / strides[dim];
```

Comment:
I think that place is better to discuss offline. It's unclear to me now why we need to divide by stride value, but compare with dimension value

Reply:
Ok, we can discuss it offline, but I'd like to notice that `row_major_strides` computes strides not by bytes but by elements (strides for f32 and for f16 will be equal). Offset is also computed by elements (`offset = [0 .. totalDimSize-1]`)

Reply:
Discussed offline.